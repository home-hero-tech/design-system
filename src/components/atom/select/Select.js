import React from 'react';
import PropTypes from 'prop-types';
import RSelect from 'react-select';

import css from './Select.module.scss';

const Select = ({
  dark,
  firstMessage,
  id,
  multiple,
  name,
  onChange,
  noOptionsMessage,
  options,
  placeholder,
  searchable,
  value,
  small,
  ...otherProps
}) => {
  let selectClass = 'c-select';

  if (dark && !small) {
    selectClass = selectClass.concat('--dark');
  } else if (!dark && small) {
    selectClass = selectClass.concat('--small');
  } else if (small && dark) {
    selectClass = selectClass.concat('--small-dark');
  } else {
    selectClass = selectClass.concat('--default');
  }

  const allProps = {
    backspaceRemoves: true,
    inputId: id || name,
    className: css[selectClass],
    classNamePrefix: selectClass,
    deleteRemoves: true,
    isMulti: multiple,
    isSearchable: searchable,
    value,
    name,
    options,
    onChange,
    placeholder,
    noOptionsMessage: ({ inputValue }) =>
      inputValue && inputValue.length ? noOptionsMessage() : firstMessage(),
    ...otherProps
  };

  return <RSelect {...allProps} />;
};

Select.propTypes = {
  dark: PropTypes.bool,
  small: PropTypes.bool,
  id: PropTypes.string,
  firstMessage: PropTypes.func,
  multiple: PropTypes.bool,
  name: PropTypes.string,
  searchable: PropTypes.bool,
  noOptionsMessage: PropTypes.func,
  onChange: PropTypes.func,
  options: PropTypes.instanceOf(Array),
  placeholder: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string,
    PropTypes.number
  ])
};

Select.defaultProps = {
  dark: false,
  small: false,
  id: null,
  firstMessage: () => 'Digite para buscar',
  multiple: false,
  name: null,
  searchable: true,
  onChange: f => f,
  options: null,
  placeholder: 'Selecionar',
  noOptionsMessage: f => f,
  value: null
};

export default Select;
