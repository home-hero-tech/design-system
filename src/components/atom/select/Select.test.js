import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from './Select';

const options = [
  { value: 'chocolate', label: 'Chocolate' },
  { value: 'strawberry', label: 'Strawberry' },
  { value: 'vanilla', label: 'Vanilla' },
  { value: 'Coconut', label: 'Coconut' }
];

const placeholder = 'Selecione...';
const noOptionsMessage = () => 'Nenhuma opção';

describe('Select', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<Select options={options} placeholder={placeholder} noOptionsMessage={noOptionsMessage} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
