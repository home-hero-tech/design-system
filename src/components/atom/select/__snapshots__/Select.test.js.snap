// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match with the snapshot 1`] = `
<StateManager
  autoFocus={false}
  className="c-select"
  classNamePrefix={null}
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  isClearable={false}
  isDisabled={false}
  isMulti={false}
  isSearchable={false}
  name={null}
  noOptionsMessage={[Function]}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Chocolate",
        "value": "chocolate",
      },
      Object {
        "label": "Strawberry",
        "value": "strawberry",
      },
      Object {
        "label": "Vanilla",
        "value": "vanilla",
      },
      Object {
        "label": "Coconut",
        "value": "Coconut",
      },
    ]
  }
  placeholder="Selecione..."
  value={null}
/>
`;
